From fbaf7f7ea8d133479da6452ddcc0a2d72baf81d4 Mon Sep 17 00:00:00 2001
From: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Date: Thu, 16 May 2013 13:27:26 +0200
Subject: [PORT FROM MAIN] ueventd/wifi: Fix possible NULL dereference

BZ: 108561

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: Ie7ad1cd03c5f41502b0701ca488b5ff2a9636c89
Orig-Change-Id: Ie6c1bfe5c53bcca83bed305f0f47f79b5159b437
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
---
 init/ueventd_parser.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/init/ueventd_parser.c b/init/ueventd_parser.c
index fe0b922..acb8369 100644
--- a/init/ueventd_parser.c
+++ b/init/ueventd_parser.c
@@ -73,6 +73,6 @@ static void parse_line_device(struct parse_state* state, int nargs, char **args)
 {
     if (args && args[0] && !strncmp("mod:", args[0], 4))
         set_module_args(nargs, args);
-    else
+    else if (args)
         set_device_permission(nargs, args);
 }
-- 
1.7.4.1

